src/game_objects/construct.cpp

```diff
@@ -12,80 +12,128 @@ Construct::Construct(int x, int z, int id) : GameObject(id) {
     transform.set_position(x, 0.0f, z);
 }
 
+void Construct::c_update_state(float x, float z, float wx, float wz) {
+    anim_player.update();
+}
+
+/************CHEST***************/
+
 Chest::Chest(int x, int z) : Construct(x, z) {
-    transform.set_scale(0.6f, 0.6f, 0.6f);
-    events::RigidBodyData rigid = {
-        glm::vec3(x,0.5f,z), //position
-        0, //mass
-        hit_box, //btshape
-        glm::vec3(0,0,0), //inertia
-        this, //the gameobject
-        false, // is a ghost object
-    };
+    tag = "CHEST";
+
+    //Creates Rigid Body
+    events::RigidBodyData rigid;
+    rigid.object = this;
+    rigid.shape = hit_box;
     events::add_rigidbody_event(rigid);
+
+    // Sets Initial Position (Transform and Rigidbody at same time)
+    set_position({ x, 0.0f, z });
```

Comment:
Not really, since the base constructor is called before the rigid body is made, and "set_position" is used to both move the game object's transform and the rigid body's transform. (I think)